dclint disable rule-name doesn't appear to be disabling rule-name

## Version

2.1.0

## YAML file where the error occurs

```YAML
---
# dclint disable require-project-name-field
# name: "does_not_matter_commented_out"
```

## Command used to run

`dclint ./docker-compose.yml`

## Optional configuration (if different from the default)

n/a

## Additional context

node: 20.18.1

## Current behavior (console output)

```
# dclint --debug ./docker-compose.yml
[DEBUG] [CLI] Debug mode is ON
[DEBUG] [CLI] Arguments: {
  _: [],
  debug: true,
  recursive: false,
  r: false,
  fix: false,
  'fix-dry-run': false,
  fixDryRun: false,
  formatter: 'stylish',
  f: 'stylish',
  quiet: false,
  q: false,
  color: true,
  exclude: [],
  e: [],
  'max-warnings': -1,
  maxWarnings: -1,
  '$0': 'dclint',
  files: [ './docker-compose.yml' ]
}
[DEBUG] [CONFIG] Configuration file not found. Using default
[DEBUG] [CLI] Final config: { rules: {}, quiet: false, debug: true, exclude: [] }
[DEBUG] [UTIL] Looking for compose files in ./docker-compose.yml
[DEBUG] [UTIL] Paths to exclude: node_modules,.git,.idea,.tsimp
[DEBUG] [UTIL] Found compose files in ./docker-compose.yml: ./docker-compose.yml
[DEBUG] [LINTER] Compose files for linting: ./docker-compose.yml
[DEBUG] [LINTER] Linting file: ./docker-compose.yml
[DEBUG] [LINTER] Linting result: [{"filePath":"./docker-compose.yml","messages":[{"rule":"require-project-name-field","type":"warning","category":"best-practice","severity":"minor","message":"The \"name\" field should be present.","line":1,"column":1,"meta":{"description":"Ensure that the \"name\" field is present in the Docker Compose file.","url":"https://github.com/zavoloklom/docker-compose-linter/blob/main/docs/rules/require-project-name-field-rule.md"},"fixable":false}],"errorCount":0,"warningCount":1,"fixableErrorCount":0,"fixableWarningCount":0}]
[DEBUG] [UTIL] Using built-in formatter: stylish

/path/to/docker-compose.yml
   1:1     warning  The "name" field should be present.  require-project-name-field

✖ 1 problems (0 errors, 1 warnings)
```

## Expected behavior

```

```

(i.e., no output because no errors or warnings were found)

## Praise

I stumbled upon dclint and I was immediately impressed by the frequency of updates, the list of PRs being clean the inclusion of linters in `.github/workflows`, the quality of the documentation, the use of common patterns to enable/disable rules, produce output, and so on.  This looks like a model project and I don't know how I haven't seen it before.

I've contributed a few things to https://github.com/oxsecurity/megalinter and my first thought was to investigate including it in the list of linters MegaLinter supports.  See oxsecurity/megalinter#4381

This isn't related to the bug report -- I just wanted to express my appreciation.

Thank you for all of your efforts and hard work.

--- a/docs/configuration-comments.md
+++ b/docs/configuration-comments.md
@@ -13,7 +13,8 @@ management across the project.
 
 ### Disabling All Rules for the Entire File
 
-To **disable all linting rules** for the entire file, add the following comment at the **top** of the file:
+To **disable all linting rules** for the entire file, add the following comment at the **first applicable line** of the
+file. This can be the first line, or the first line after a YAML separator (`---`):
 
 ```yaml
 # dclint disable-file
@@ -23,7 +24,8 @@ This will disable **all rules** for the entire file. Use this only when it is ab
 
 ### Disabling Specific Rules for the Entire File
 
-To **disable specific rules** for the entire file, add the following comment at the **top** of the file:
+To **disable specific rules** for the entire file, add the following comment at the **first applicable line**. This can
+be the first line, or the first line after a YAML separator (`---`):
 
 ```yaml
 # dclint disable

--- a/src/util/comments-handler.ts
+++ b/src/util/comments-handler.ts
@@ -1,25 +1,35 @@
 function startsWithDisableFileComment(content: string): boolean {
-  return content.startsWith('# dclint disable-file');
+  // Split content into lines and process the first non-empty, non-YAML separator line
+  const lines = content.split('\n').map((line) => line.trim());
+  for (const line of lines) {
+    // eslint-disable-next-line no-continue
+    if (line === '' || line === '---') continue;
+    return line.startsWith('# dclint disable-file');
+  }
+  return false;
 }
 
 function extractGlobalDisableRules(content: string): Set<string> {
   const disableRules = new Set<string>();
 
-  // Get the first line and trim whitespace
-  const firstLine = content.trim().split('\n')[0].trim();
+  // Split content into lines and process the first non-empty, non-YAML separator line
+  const lines = content.split('\n').map((line) => line.trim());
+  for (const line of lines) {
+    // eslint-disable-next-line no-continue
+    if (line === '' || line === '---') continue;
+    const disableMatch = line.match(/#\s*dclint\s+disable\s*(.*)/);
+    if (disableMatch) {
+      const rules = disableMatch[1].trim();
 
-  // Check if the first line contains "dclint disable"
-  const disableMatch = firstLine.match(/#\s*dclint\s+disable\s*(.*)/);
-  if (disableMatch) {
-    const rules = disableMatch[1].trim();
-
-    // If no specific rules are provided, disable all rules
-    if (rules === '') {
-      disableRules.add('*');
-    } else {
-      // Otherwise, disable specific rules mentioned
-      rules.split(/\s+/).forEach((rule) => disableRules.add(rule));
+      // If no specific rules are provided, disable all rules
+      if (rules === '') {
+        disableRules.add('*');
+      } else {
+        // Otherwise, disable specific rules mentioned
+        rules.split(/\s+/).forEach((rule) => disableRules.add(rule));
+      }
     }
+    break;
   }
 
   return disableRules;

--- a/tests/util/comments-handler.spec.ts
+++ b/tests/util/comments-handler.spec.ts
@@ -16,6 +16,15 @@ test('startsWithDisableFileComment should return true when content starts with "
 });
 
 // @ts-ignore TS2349
+test('startsWithDisableFileComment should return true when content starts with delimiter and "# dclint disable-file"', (t) => {
+  const content = '---\n\n# dclint disable-file\nversion: "3"';
+  t.true(
+    startsWithDisableFileComment(content),
+    'Function should return true if content starts with delimiter and "# dclint disable-file"',
+  );
+});
+
+// @ts-ignore TS2349
 test('startsWithDisableFileComment should return false when content does not start with "# dclint disable-file"', (t) => {
   const content = 'version: "3"\n# dclint disable-file';
   t.false(
@@ -51,6 +60,23 @@ test('extractGlobalDisableRules should disable all rules if # dclint disable is 
     [...result],
     ['*'], // '*' means all rules disabled
     'Should disable all rules for the entire file (comment in the first line)',
+  );
+});
+
+// @ts-ignore TS2349
+test('extractGlobalDisableRules should find disable rules after delimiter', (t) => {
+  const content = `
+    ---
+    
+    # dclint disable
+    key: value 1
+    key: value 2
+  `;
+  const result = extractGlobalDisableRules(content);
+  t.deepEqual(
+    [...result],
+    ['*'], // '*' means all rules disabled
+    'Should disable all rules for the entire file (comment in the line after delimiter)',
   );
 });
 

